File: frontend/src/api/endpoints/imagery.ts
import type { ApiClient } from '../client';
import type { ImageryFrame } from '@/types/imagery';
import type { SensorType } from '@/types/sensor';

export interface ImageryQuery {
  bbox?: [number, number, number, number];
  since?: string;
  until?: string;
  source?: SensorType;
  limit?: number;
  offset?: number;
}

export const bind = (client: ApiClient) => ({
  list: (_query?: ImageryQuery): Promise<ImageryFrame[]> =>
    client.get('/imagery', { query: _query as never }),
  signedUrl: (id: string): Promise<{ url: string }> => client.get(`/imagery/${id}/url`),
});

File: backend/app/schemas/imagery.py
"""Imagery DTOs."""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.domain.enums import SensorType
from app.schemas.common import GeoPointDTO


class ImageryFrameRead(BaseModel):
    id: UUID
    source: SensorType
    sensor_id: UUID
    captured_at: datetime
    centre: GeoPointDTO
    footprint: list[GeoPointDTO]
    resolution_m: float
    storage_uri: str


class ImageryQuery(BaseModel):
    bbox: tuple[float, float, float, float] | None = None
    since: datetime | None = None
    until: datetime | None = None
    source: SensorType | None = None
    limit: int = 50
    offset: int = 0
